tooling/validate-registry.js

```diff
@@ -0,0 +1,47 @@
+
+const Ajv = require("ajv")
+const yaml = require('js-yaml');
+const fs   = require('fs');
+const path   = require('path');
+const ajv = new Ajv({ strict: false });
+
+const didMethodRegistryDirectory = path.join(__dirname, '../methods');
+ 
+const schema = yaml.load(fs.readFileSync('./did-method-registry-entry.yml', 'utf8'));
+const validate = ajv.compile(schema)
+  
+const getAllRegistryEntries = () =>{
+    const files = fs.readdirSync(didMethodRegistryDirectory);
+    const entries = files.filter((file)=>{
+        // ignore the index file.
+        return file !== 'index.json';
+    }).map((file) => {
+        const fileContent = fs.readFileSync(path.join(didMethodRegistryDirectory, file)).toString();
+        return JSON.parse(fileContent);
+    }).sort((a, b)=>{
+        return a.name > b.name ? 1 : -1;
+    })
+    return entries
+}
+
+const validateRegistryEntry = (entry)=>{
+    const valid = validate(entry)
+    if (!valid) {
+        console.error(entry)
+        console.error(validate.errors)
+    }
+    return valid;
+}
+
+(async ()=>{
+    console.log('🧙  validating did method registry');
+    const entries = getAllRegistryEntries();
```

Comment:
`getAllRegistryEntries` can throw (e.g., failed JSON parse) -- dunno if we care about wrapping any of this in a try/catch or not, just noting it.

Reply:
Agree, we want to catch and report the error because people are going to submit broken JSON and we don't want the tool to fall over when that happens.

Reply:
Fixed in latest set of commits.